core/src/main/scala/cats/data/Streaming.scala

```diff
@@ -523,16 +524,17 @@ sealed abstract class Streaming[A] extends Product with Serializable { lhs =>
    * If no elements satisfy `f`, the current stream will be returned.
    *
    * For example:
-   *
-   *   Streaming(1, 2, 3, 4, 5, 6, 7).takeWhile(n => n != 4)
-   *
-   * Will result in: Streaming(4, 5, 6, 7)
+   * {{{
+   * scala> val s = Streaming(1, 2, 3, 4, 5, 6, 7)
+   * scala> s.dropWhile(n => n != 4).toList
+   * res0: List[Int] = List(4, 5, 6, 7)
+   * }}}
    */
   def dropWhile(f: A => Boolean): Streaming[A] =
     this match {
       case Empty() => Empty()
       case Wait(lt) => Wait(lt.map(_.dropWhile(f)))
-      case Cons(a, lt) => if (f(a)) Empty() else Cons(a, lt.map(_.dropWhile(f)))
+      case Cons(a, lt) => if (f(a)) Wait(lt.map(_.dropWhile(f))) else Cons(a, lt)
```

Comment:
Good spot - Thanks! - In general, is there any reason to do `s @` instead and evaluate the expression to `s` instead of just evaluating to `this`? Aesthetically I prefer using `this`, but that is just personal preference - was wondering if there is any real reason I not aware of to prefer one of the other. Eitherways, I'll push through with the `s @` ; because the only reason I have any preference for using `this` is aesthetic, and that likely comes from Java/C++